File: Solved/Chape Baraks/Chape Baraks.py
# چاپ برعکس
# ID : 3405
# https://quera.ir/problemset/contest/3405/%D8%B3%D8%A4%D8%A7%D9%84-%DA%86%D8%A7%D9%BE-%D8%A8%D8%B1%D8%B9%DA%A9%D8%B3
# https://b2n.ir/637768

lister = []
while True:
    number = input()
    if number == "0":
        break
    lister.append(number)
for i in lister[::-1]:
    print(i)

File: Solved/Chape Baraks/ChapeBaraks.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

/*
Question title = chap barax - چاپ برعکس
https://quera.ir/problemset/3405/
 */

public class ChapeBaraks {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<String> list = new ArrayList<>();
        while (true) {
            String input = sc.nextLine().trim();
            if (input.equals("0")) {
                break;
            } else {
                list.add(input);
            }
        }
        Collections.reverse(list);
        for (String i : list) {
            System.out.println(i);
        }
    }
}